File: kubik/kubik-server/src/main/java/com/cspinformatique/kubik/server/domain/purchase/service/PurchaseOrderService.java
package com.cspinformatique.kubik.server.domain.purchase.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import com.cspinformatique.kubik.server.model.product.Product;
import com.cspinformatique.kubik.server.model.product.Supplier;
import com.cspinformatique.kubik.server.model.purchase.PurchaseOrder;
import com.cspinformatique.kubik.server.model.purchase.Reception;
import com.cspinformatique.kubik.server.model.purchase.PurchaseOrder.Status;

public interface PurchaseOrderService {
	
	void confirmAllOrders();
	
	Iterable<PurchaseOrder> findAll();
	
	Page<PurchaseOrder> findAll(Pageable pageable);
	
	List<PurchaseOrder> findByProduct(Product product);
	
	List<PurchaseOrder> findByProductAndStatus(Product product, Status status);
	
	List<PurchaseOrder> findByStatus(Status status);
	
	PurchaseOrder findOne(long id);
	
	Reception generateReception(PurchaseOrder purchaseOrder);
	
	void fixSubmitedDate();
	
	void recalculateOpenPurchaseOrderFromSupplier(Supplier supplier);
	
	PurchaseOrder save(PurchaseOrder purchaseOrder);
	
	Iterable<PurchaseOrder> save(Iterable<PurchaseOrder> entities);
}

File: kubik/kubik-server/src/main/webapp/public/js/purchase/purchase-order-service.js
(function(){
	var PURCHASE_ORDER_URL = "/purchaseOrder"
	angular
		.module("Kubik")
		.factory("purchaseOrderService", PurchaseOrderService);
	
	function PurchaseOrderService($http){
		return {
			findOne : findOne
		};
		
		function findOne(purchaseOrderId){
			return $http
				.get(PURCHASE_ORDER_URL + "/" + purchaseOrderId)
				.then(findOneSuccess);
			
			function findOneSuccess(response){
				return response.data;
			}			
		}
	}
})();